microcosm_pubsub/producer.py

```diff
@@ -109,6 +111,33 @@ def __exit__(self, type, value, traceback):
             self.producer.publish_message(media_type, message, topic_arn, opaque_data)
 
 
+def collapse_dict(dct, prefix="", separator="."):
+    """
+    Collapse a nested dictionary into a single-level dictionary.
+
+    Since "." is not legal in an environment variable, we can't easily express our
+    pubsub overrides in environment variable form. The only legal special characters
+    is underscore and our configuration loader already uses double underscore as
+    a dictionary separator. So we (re)collapse nested dictionaries here.
+
+    """
+    for key, value in dct.items():
+        if isinstance(value, dict):
+            for nested_key, nested_value in collapse_dict(value, key, separator):
+                yield separator.join([prefix, nested_key]), nested_value
+        else:
+            yield separator.join([prefix, key]), value
+
+
+def iter_topic_mappings(dct):
+    for key, value in dct.items():
+        if isinstance(value, string_types):
+            yield key, value
+        else:
+            for key, value in collapse_dict(value, key):
```

Comment:
You can't return from inside a generator; you have to yield.

I fixed the naming.